helper/threads.c

```diff
@@ -407,6 +419,34 @@ int odph_thread_join(odph_thread_t thread[], int num)
 	return i;
 }
 
+int odph_thread_join_result(odph_thread_t thread[], odph_thread_join_result_t res[], int num)
+{
+	odph_thread_start_args_t *start_args;
+	int num_exited = 0;
+
+	for (int i = 0; i < num; i++) {
+		start_args = &thread[i].start_args;
+
+		if (start_args->status != STARTED) {
+			ODPH_DBG("Thread (i:%i) not started.\n", i);
+			continue;
+		}
```

Comment:
This may cause usage of uninitialized variable warnings as some `res` elements are not touched. Also, thinking whether this should be also considered as a failure -> break/return.

Reply:
Since this is not a fast path function I'd keep the extra check as it can help in debugging user errors.